### PR TITLE
Rework user service

### PR DESCRIPTION
**Pull request by jerben**

### CODE REVIEW

File: sihl/src/contract_user.ml [hunk 1]
```diff
@@ -101,73 +126,79 @@ module type Sig = sig
...
+  (** [create_admin ?username ?name ?given_name email password] returns an admin
+      user. *)
   val create_admin
-    :  email:string
-    -> password:string
-    -> username:string option
+    :  ?username:string
+    -> ?name:string
+    -> ?given_name:string
+    -> string
```
Comment (aronerben) [inline sihl/src/contract_user.ml:146]: Why are these not named anymore? They are both `string` and are next to each other

File: sihl-user/src/sihl_user.ml [hunk 2]
```diff
@@ -17,11 +17,6 @@ module Make (Repo : User_repo.Sig) : Sihl.Contract.User.Sig = struct
   ;;
 
   let find_by_email_opt ~email =
-    (* TODO add support for lowercase UTF-8
```
[the remark this answers is not in the document]
Comment (jerben) [reply sihl-user/src/sihl_user.ml:20]: No I fixed it by dealing with it on repo level.

File: sihl/src/contract_user.ml (hunk 1 above)
Comment (jerben) [reply sihl/src/contract_user.ml:146]: Hmm true, maybe we can have email as `string` next to `password:string`? If passwords get stored as plaintext that is quite bad.

File: sihl-user/src/sihl_user.ml (hunk 2 above)
[the remark this answers is not in the document]
Comment (aronerben) [reply sihl-user/src/sihl_user.ml:20]: I think this comment refers to something else. `String.lowercase_ascii` will (probably) generate gibberish when it is used on non-ASCII email addresses. So the repo might get a complete garbage mail to look for if for example there was an umlaut in the mail that we just tried to lowercase.

File: sihl/src/contract_user.ml (hunk 1 above)
Comment (aronerben) [reply sihl/src/contract_user.ml:146]: Yea that sounds good

File: sihl-user/src/sihl_user.ml (hunk 2 above)
[the remark this answers is not in the document]
Comment (jerben) [reply sihl-user/src/sihl_user.ml:20]: `WHERE LOWER(user_users.email) = LOWER(?)`

[the remark this answers is not in the document]
Comment (jerben) [reply sihl-user/src/sihl_user.ml:20]: Ah I just saw that `String.lowercase_ascii` is still there, let me remove it quickly!

[the remark this answers is not in the document]
Comment (aronerben) [reply sihl-user/src/sihl_user.ml:20]: Thanks!

File: sihl/src/contract_user.ml (hunk 1 above)
Comment (jerben) [reply sihl/src/contract_user.ml]: Fixed.

Comment (aronerben) [reply sihl/src/contract_user.ml]: Thanks

**Comment by aronerben** [comment]
CI failing FYI

**Pull request merged**